Runtime/Plugins/webgl/StandaloneFileBrowser.jslib

```diff
@@ -114,20 +114,59 @@ mergeInto(LibraryManager.library, {
             };
 
             var transaction = window.databaseConnection.transaction(["FILE_DATA"], "readwrite");
-            var newIndexedFilePath = window.databaseName + "/" + filename;
-            var dbRequest = transaction.objectStore("FILE_DATA").put(data, newIndexedFilePath);
-
-            console.log("Saving file: " + newIndexedFilePath);
-            dbRequest.onsuccess = function () {
-                SendMessage('UserFileUploads', 'LoadFile', filename);
-                console.log("File saved: " + newIndexedFilePath);
-                window.FileSaved();
-            };
-            dbRequest.onerror = function () {
-                SendMessage('UserFileUploads', 'LoadFileError', filename);
-                alert("Could not save: " + newIndexedFilePath);
-                window.FileSaved();
-            };
+            var objectStore = transaction.objectStore("FILE_DATA");//.put(data, newIndexedFilePath);
+
+            function getUniqueFileName(objectStore, baseName, extension, callback) {
+                function buildFileName(base, number, ext) {
+                    return number > 0 ? `${base}(${number})${ext}` : `${base}${ext}`;
+                }
+
+                let number = 0;
+                let fileName = buildFileName(baseName, number, extension);
+
+                function checkFileName() {
+                    let key = window.databaseName + "/" + fileName;
+                    let request = objectStore.getKey(key);  // Use getKey to check if the file exists
+
+                    request.onsuccess = function () {
+                        if (request.result !== undefined) {
+                            number++;
+                            fileName = buildFileName(baseName, number, extension);
+                            checkFileName();  // Check the next possible file name
```

Comment:
moet die callback hier niet ook iets returnen of betekent dit echt alleen callback doen voor unieke files

Reply:
hij gaat recursive incrementen, dus die callback komt uiteindelijk wel